rotkehlchen/chain/evm/decoding/paraswap/decoder.py

```diff
@@ -0,0 +1,202 @@
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.FEE,
+                asset=fee_asset,
+                balance=Balance(amount=fee_amount),
+                location_label=sender,
+                notes=f'Spend {fee_amount} {fee_asset.symbol} as a paraswap fees',
+                counterparty=CPT_PARASWAP,
+                address=context.transaction.to_address,
+            )
+            context.decoded_events.append(fee_event)
+
+        maybe_reshuffle_events(
+            ordered_events=[out_event, in_event, fee_event],
+            events_list=context.decoded_events,
+        )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_paraswap_swap(self, context: DecoderContext) -> DecodingOutput:
+        """This decodes the following types of trades:
+        - Simple Buy
+        - Simple Swap
+        - Multi Swap
+        - Mega Swap"""
+        if context.tx_log.topics[0] not in {PARASWAP_SWAP_SIGNATURE, BUY_SIGNATURE}:
+            return DEFAULT_DECODING_OUTPUT
+
+        return self._decode_swap(
+            context=context,
+            receiver=hex_or_bytes_to_address(context.tx_log.topics[1]),
+            sender=hex_or_bytes_to_address(context.tx_log.data[96:128]),
+        )
+
+    def _decode_uniswap_v2_swap(self, context: DecoderContext) -> DecodingOutput:
+        """This decodes swaps done directly on Uniswap V2 pools"""
+        return self._decode_swap(
+            context=context,
+            receiver=hex_or_bytes_to_address(context.tx_log.topics[2]),
+            sender=context.transaction.from_address,
+        )
+
+    def _decode_uniswap_v3_swap(self, context: DecoderContext) -> DecodingOutput:
+        """This decodes swaps done directly on Uniswap V3 pools"""
+        return self._decode_swap(
+            context=context,
+            receiver=hex_or_bytes_to_address(context.tx_log.topics[1]),
+            sender=hex_or_bytes_to_address(context.tx_log.data[96:128]),
+        )
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            self.router_address: (self._decode_paraswap_swap,),
+        }
+
+    def decoding_by_input_data(self) -> dict[bytes, dict[bytes, Callable]]:
+        return {
+            b'\x0b\x86\xa4\xc1': {  # swapOnUniswapV2Fork
+                UNISWAP_V2_SWAP_SIGNATURE: self._decode_uniswap_v2_swap,
+            },
+            b'n\x91S\x8b': {  # swapOnUniswapV2ForkWithPermit
+                UNISWAP_V2_SWAP_SIGNATURE: self._decode_uniswap_v2_swap,
+            },
+            b'\xa6\x88m\xa9': {  # directUniV3Swap
+                DIRECT_SWAP_SIGNATURE: self._decode_uniswap_v3_swap,
+            },
+        }
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (CounterpartyDetails(
+            identifier=CPT_PARASWAP,
+            label='paraswap swaps',
```

Comment:
This is the label shown for the protocol. The protocol is paraswap. Don't copy the same as metamask. Metamask was a special case as it's a wallet (Metamask) that also created their own swaps, hence the protocol was called Metamask Swaps.

Also since label is frontend you can capitalize.

```suggestion
            label='Paraswap',
```